java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -251,7 +251,7 @@ public static PackageChangeOutcome handlePackageChanges(String function,
             ErrataManager.insertErrataCacheTask(server);
             outcome = applyChangesFromStateModule(delta, server);
         }
-        else if (function.equals("state.apply")) {
+        else if (function != null && function.equals("state.apply")) {
```

Comment:
Did you had the chance to test this and see that it does not fail anymore?

Anyway, since the `String.equals` allows receiving `null` arguments, why not just reverting the order of the check instead of adding another one? `else if ("state.apply".equals(function))`

Reply:
@ncounter I will deploy my branch to refhead and see if it works

Reply:
The fix works but now there is another error. It expects the wrong text. Is this another PR or shall I fix that also?

Reply:
If the error is different and the line affected is different, it's definitely a different PR, even a different bug I'd say. What is failing and how?

Reply:
At least, the fix deserve a separate commit with the proper bugnumber in the commit text. If you want to add the second fix here, let's do it but keep it in different commits.

Reply:
Okay then I will investigate deeper and decide then if this error deserves an own PR or if I just do a commit.

Reply:
why is `function` null here? The method clearly expects a String and passing anything else to it is invalid use. There is a lot of other functions in this file and probably around the whole codebase taking in the exact same `function` variable expecting it to be a String. This is not solved by adding a null check here but by finding out where the null comes from and properly marking the source as `Optional`. This way all places directly or indirectly getting function from this source will become compile errors until the `Optional` is handled.

Reply:
@SchoolGuy I remember you told me you based this fix on the assumption that it is fine to have a `null` function value here (did you discussed it with some Salt guy?). Maybe this assumption needs to be revisited/dropped according to @lucidd 's comment which is definitely correct. If the `function` key is evaluated in many places and expected to be `String`, we definitely need something different than this check.

Reply:
@ncounter @lucidd Then I will investigate deeper and try to find the real cause.